src/main/user-impl/java/com/mysql/cj/jdbc/ha/plugins/failover/AuroraTopologyService.java

```diff
@@ -98,18 +97,20 @@
 
   /** Initializes a service with topology default refresh rate. */
   public AuroraTopologyService(Log log) {
-    this(DEFAULT_REFRESH_RATE_IN_MILLISECONDS, log);
+    this(DEFAULT_REFRESH_RATE_IN_MILLISECONDS, log, () -> new ClusterAwareMetricsContainer(null, null));
```

Comment:
There's no null checks inside ClusterAwareMetricsContainer for currentConnectionProvider. It's unsafe.

Reply:
Default constructor of ClusterAwareMetricsContainer would look better here.